CMakeLists.txt

```diff
@@ -12,14 +12,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cmake_minimum_required(VERSION 3.12)
+cmake_minimum_required(VERSION 3.13.4)
 project(sxg)
 
 set(CMAKE_PROJECT_VERSION "0")
 set(CMAKE_PROJECT_VERSION_MAJOR "0")
 set(CMAKE_PROJECT_VERSION_MINOR "1")
 set(CMAKE_PROJECT_VERSION_PATCH "0")
 
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Release)
+endif()
+if(NOT DEFINED WITH_TEST)
```

Comment:
Maybe it can be `SKIP_TEST` which is naturally defaulted false, especially if skipping unit tests is unusual except for `make install` (to _me_ `SKIP_TEST=yes` looks more unusual than `WITH_TEST=no`).

Otherwise I suggest `ENABLE_TEST`. `WITH_TEST=on` sounds a bit unnatural though `WITH_TEST=yes` sounds okay.

Reply:
OK, `SKIP_TEST=TRUE` will work.